### PR TITLE
ISSUE-90: Add library sample

### LINKED ISSUE #90: Update docs/sample to include how to run in non-app modules (e.g. Android library module)
**Is your feature request related to a problem? Please describe.**
I'd like to know how to get the tests working in a non-app module such as an Android library module. Would be fantastic if the docs mentioned how to do this, and even better some sample code to demonstrate it.

This feedback relates to:
- [x] The Kotlin library
- [x] The Gradle plugin
- [ ] The IntelliJ Platform plugin
- [x] The sample code
- [x] The documentation

**Describe the solution you'd like**
As mentioned above.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

---

_Copied from @ryansama https://github.com/Shopify/android-testify/issues/272_

### PR DESCRIPTION
**Pull request by DanielJette**
### What does this change accomplish?

Resolves #90

# Configuring Testify for Android Library Projects

An [Android library](https://developer.android.com/studio/projects/android-library) is structurally the same as an Android app module. It includes everything needed to build an app, including source code, resource files, and an Android manifest. So, it is often necessary to want to use screenshot testing with your library. 

However, instead of compiling into an APK that runs on a device, an Android library compiles into an Android Archive (AAR) file that you can use as a dependency for an Android app module. This means that Testify requires special configuration. The Testify Gradle Plugin is able to infer most configuration parameters for App modules, but for a Library module, some values must be specified in the `build.gradle` file.

In particular, unlike an App module, a Library project does not produce an APK of its own. Therefore, Testify must be configured to use the Library's generated instrumentation test runner APK as both the application under test and the test runner. These values are specified via the `applicationPackageId` and `testPackageId` Testify Extension values.

The `applicationPackageId` and `testPackageId` for a Library project will be the same value. You must specify this in your `build.gradle` file:

```groovy
testify {
    applicationPackageId "dev.testify.samples.flix.library.test"
    testPackageId "dev.testify.samples.flix.library.test"
}
```

To find out what value to use for `applicationPackageId` and `testPackageId`, you can check the [_instrumentation_](https://source.android.com/docs/core/tests/development/instrumentation) name for your library by using the `pm list instrumentation` command on your emulator.

For example, to check the instrumentation name of the _Flix Library_ sample project:

```bash
~>: ./gradlew FlixLibrary:installDebugAndroidTest
    Installing APK 'FlixLibrary-debug-androidTest.apk' on 'Testify(AVD) - 10' for :FlixLibrary:debug-androidTest
    BUILD SUCCESSFUL in 1s
    53 actionable tasks: 1 executed, 52 up-to-date

~>: adb shell pm list instrumentation
    instrumentation:dev.testify.samples.flix.library.test/androidx.test.runner.AndroidJUnitRunner (target=dev.testify.samples.flix.library.test)
```

#### Testify Configuration Values:

- **testPackageId** : The `instrumentation` package reported by `pm list instrumentation`.<br/>
    ex. `dev.testify.samples.flix.library.test`

- **applicationPackageId** : The `target` package reported by `pm list instrumentation`.<br/>
    ex. `dev.testify.samples.flix.library.test`

---

### Sample

Please see the [Flix Library sample](https://github.com/ndtp/android-testify/tree/main/Samples/Flix/FlixLibrary) for an example of using Testify with a library project.

### More Information

- [Create an Android library](https://developer.android.com/studio/projects/android-library)
- [Instrumentation tests](https://source.android.com/docs/core/tests/development/instrumentation)

---

### Notice

> [!WARNING]
> This change must keep `main` in a shippable state; **it may be shipped without further notice**.

### CODE REVIEW

File: Samples/Flix/FlixLibrary/proguard-rules.pro [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Add project specific ProGuard rules here.
```
Comment (lwise) [inline Samples/Flix/FlixLibrary/proguard-rules.pro:1]: nit: delete these?

File: bitrise.yml [hunk 2]
```diff
@@ -167,9 +163,6 @@ workflows:
         inputs:
         - test_name: Flix Screenshot Test
         - search_pattern: "**test_result_**.xml**"
-    - deploy-to-bitrise-io@2:
```
Comment (lwise) [inline bitrise.yml:170]: Why did this get removed?

**Review by lwise — APPROVED** [review]

File: Samples/Flix/FlixLibrary/proguard-rules.pro (hunk 1 above)
Comment (DanielJette) [reply Samples/Flix/FlixLibrary/proguard-rules.pro:1]: Argh. Meant to delete these, yes. Will remove

File: bitrise.yml (hunk 2 above)
Comment (DanielJette) [reply bitrise.yml:170]: @lwise There were too many of these. They copy the artifacts to the `Artifacts` tab on the Bitrise report but there only needs to be one of these. There's one on L259

**Pull request merged**